src/bson.c

```diff
@@ -885,38 +885,38 @@ int bson_to_zval(const unsigned char *data, int data_len, php_phongo_bson_state
 
 	reader = bson_reader_new_from_data(data, data_len);
 
-	if (!(b = bson_reader_read(reader, &eof))) {
+	if (!(b = bson_reader_read(reader, NULL))) {
```

Comment:
I'm passing `NULL` here because we don't actually check `eof` until the second call, which your original do/while loop was accomplishing.